src/components/SearchPage.tsx

```diff
@@ -3,36 +3,26 @@ import Card from './Card'
 import Pagination from './Pagination'
 import { fetchCards, ListCardsResponse } from 'utils'
 import { useFormInputDebounce } from 'hooks'
+import { useQuery } from 'react-query'
+
+const getCards = (name: string, currentPage: number) => {
+  return fetchCards(`&page=${currentPage}&q=name:${name}*`)
+}
 
 const SearchPage = () => {
   const inputRef = useRef<HTMLInputElement>(null)
 
   const [currentPage, setCurrentPage] = useState(1)
-  const [isLoaded, setIsLoaded] = useState(false)
-  const [isError, setIsError] = useState(false)
+  const [searchName, setSearchName] = useState('')
   const [formData, setFormData] = useState({
     pokemonName: '',
   })
-  const [responseData, setResponseData] = useState<ListCardsResponse>({
-    data: [],
-    page: 1,
-    pageSize: 1,
-    count: 1,
-    totalCount: 0,
-  })
+  
+  const { isLoading, isError, data } = useQuery<ListCardsResponse, Error>(
```

Comment:
it's best practice to not destructure these variables actually, if there's multiple queries in 1 component it becomes a nightmare quickly if you're destructuring all these different variables.